go/vt/vttablet/tabletserver/vstreamer/rowstreamer.go

```diff
@@ -200,6 +200,8 @@ func (rs *rowStreamer) buildSelect() (string, error) {
 		buf.Myprintf("%s%v", prefix, sqlparser.NewColIdent(rs.plan.Table.Fields[pk].Name))
 		prefix = ", "
 	}
+
+	buf.Myprintf(" LIMIT 1000000")
```

Comment:
good for the next experiment but i suspect this to no be mergeable upstream without a configuration flag for it. we can cross that bridge after we confirm this works.